base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorView.java

```diff
@@ -288,13 +288,13 @@ private void displayLoaded() {
 			    // Make buttons to move an action up and down the list
 		        Button btnMoveActionUp = createMoveRowButton(moveComposite, "move_up.png", "up");
 		        btnMoveActionUp.addListener(SWT.Selection, e -> {
-		        	scriptGeneratorViewModel.moveActionUp(table.getSelectionIndex());
+		        	scriptGeneratorViewModel.moveActionUp(table.getSelectionIndices());
 		        	table.setSelectionIndex(Math.max(0, table.getSelectionIndex() - 1));
```

Comment:
There should be some code here that gets the original selection, moves it up one providing the selection isn't already at the top (at index 0), and then sets that as the new selection once `moveActionUp()` has been called. The same should be down for the move button down listener below.